From e355017b2167a39b1d77959d0c9d3ee63022c7f5 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Fri, 12 Sep 2014 01:56:39 -0400
Subject: [drm] i915: add some registers need for displayport MST support

Message-id: <1410487019-24661-2-git-send-email-airlied@redhat.com>
Patchwork-id: 93912
O-Subject: [RHEL7.1 BZ1140440 PATCH 01/21] drm/i915: add some registers need for displayport MST support.
Bugzilla: 1140440
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Rob Clark <rclark@redhat.com>

bugzilla: 1140440
upstream: 01b887c36e12845bb9162e65504de623f1a17f10

These are just from the Haswell spec.

Signed-off-by: Dave Airlie <airlied@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index a5bab61..f548101 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -5743,6 +5743,7 @@ enum punit_power_well {
 #define  TRANS_DDI_EDP_INPUT_A_ONOFF (4<<12)
 #define  TRANS_DDI_EDP_INPUT_B_ONOFF (5<<12)
 #define  TRANS_DDI_EDP_INPUT_C_ONOFF (6<<12)
+#define  TRANS_DDI_DP_VC_PAYLOAD_ALLOC (1<<8)
 #define  TRANS_DDI_BFI_ENABLE  (1<<4)
 
 /* DisplayPort Transport Control */
@@ -5752,6 +5753,7 @@ enum punit_power_well {
 #define  DP_TP_CTL_ENABLE   (1<<31)
 #define  DP_TP_CTL_MODE_SST   (0<<27)
 #define  DP_TP_CTL_MODE_MST   (1<<27)
+#define  DP_TP_CTL_FORCE_ACT   (1<<25)
 #define  DP_TP_CTL_ENHANCED_FRAME_ENABLE (1<<18)
 #define  DP_TP_CTL_FDI_AUTOTRAIN  (1<<15)
 #define  DP_TP_CTL_LINK_TRAIN_MASK  (7<<8)
@@ -5766,8 +5768,13 @@ enum punit_power_well {
 #define DP_TP_STATUS_A   0x64044
 #define DP_TP_STATUS_B   0x64144
 #define DP_TP_STATUS(port) _PORT(port, DP_TP_STATUS_A, DP_TP_STATUS_B)
-#define  DP_TP_STATUS_IDLE_DONE  (1<<25)
-#define  DP_TP_STATUS_AUTOTRAIN_DONE (1<<12)
+#define  DP_TP_STATUS_IDLE_DONE   (1<<25)
+#define  DP_TP_STATUS_ACT_SENT   (1<<24)
+#define  DP_TP_STATUS_MODE_STATUS_MST  (1<<23)
+#define  DP_TP_STATUS_AUTOTRAIN_DONE  (1<<12)
+#define  DP_TP_STATUS_PAYLOAD_MAPPING_VC2 (3 << 8)
+#define  DP_TP_STATUS_PAYLOAD_MAPPING_VC1 (3 << 4)
+#define  DP_TP_STATUS_PAYLOAD_MAPPING_VC0 (3 << 0)
 
 /* DDI Buffer Control */
 #define DDI_BUF_CTL_A    0x64000
-- 
1.7.1